jobs/client.go

```diff
@@ -0,0 +1,67 @@
+package jobs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	l "github.com/RedHatInsights/sources-api-go/logger"
+	"github.com/RedHatInsights/sources-api-go/redis"
+)
+
+type JobRequest struct {
+	JobName string
+	JobRaw  []byte
+	Job     Job
+}
+
+// implementing binary mashaler/unmarshaler interfaces for redis encoding/decoding.
+func (jr JobRequest) MarshalBinary() (data []byte, err error) {
+	return json.Marshal(&jr)
+}
+func (jr *JobRequest) UnmarshalBinary(data []byte) error {
+	return json.Unmarshal(data, jr)
+}
+
+func (jr *JobRequest) Parse() error {
+	switch jr.JobName {
+	case "SuperkeyDestroyJob":
+		sdj := SuperkeyDestroyJob{}
+		err := json.Unmarshal([]byte(jr.JobRaw), &sdj)
```

Comment:
I think casting the `JobRaw` to `[]byte` is redundant here

Reply:
Yep good catch that is leftover from when I had the field as a string instead of a []byte.